src/app/webui/next.config.ts

```diff
@@ -0,0 +1,38 @@
+import type { NextConfig } from 'next';
+import os from 'os';
+
+// Determine allowed development origins (local network IPs on port 3000)
+const interfaces = os.networkInterfaces();
+const allowedOrigins: string[] = ['http://localhost:3000'];
+Object.values(interfaces).forEach((list) =>
+    list?.forEach((iface) => {
+        if (iface.family === 'IPv4' && !iface.internal) {
+            allowedOrigins.push(`http://${iface.address}:3000`);
+        }
+    })
+);
+const nextConfig: NextConfig = {
+    reactStrictMode: true,
+    // Allow static asset requests from these origins in dev mode
+    allowedDevOrigins: allowedOrigins,
+    async rewrites() {
+        const apiPort = process.env.API_PORT ?? '3001';
```

Comment:
we should update src/app/index.ts to also support `process.env.API_PORT` otherwise if this env variable is passed it wont work correctly

Reply:
can standardize all env variables for ports/server URLs separately in a different PR